lib/ansible/modules/network/ftd/ftd_install.py

```diff
@@ -0,0 +1,305 @@
+#!/usr/bin/python
+
+# Copyright (c) 2019 Cisco and/or its affiliates.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: ftd_install
+short_description: Installs FTD pkg image on the firewall
+description:
+  - Provisioning module for FTD devices that installs ROMMON image (if needed) and
+    FTD pkg image on the firewall.
+  - Can be used with `httpapi` and `local` connection types. The `httpapi` is preferred,
+    the `local` connection should be used only when the device cannot be accessed via
+    REST API.
+version_added: "2.8"
+requirements: [ "python >= 3.5", "kick" ]
+author: "Cisco Systems, Inc. (@annikulin)"
+options:
+  device_hostname:
```

Comment:
Should some of these parameters be located in the module utility and shared with other FTD modules?

Reply:
Good point, but currently we don't have other FTD modules with similar parameters. The rest of FTD modules (e.g., `ftd_configuration`, `ftd_file_download`, `ftd_file_upload`) use the Ansible HTTP API connection with the FTD plugin, so username and password are specified as a part of inventory file.
`ftd_install`, on the contrary, has them as module params, because it should do image installation even on devices without API (e.g., when the device is broken or the previous image didn't have API exposed). Does it make sense?

Reply:
That does make sense. Thanks.